src/app/credExplorer/WeightConfig.js

```diff
@@ -65,26 +61,49 @@ export class WeightConfig extends React.Component<Props, State> {
               justifyContent: "space-between",
             }}
           >
-            <EdgeConfig
-              edgeWeights={this.state.edgeWeights}
-              onChange={(ew) =>
-                this.setState({edgeWeights: ew}, () => this.fire())
-              }
-            />
-            <NodeConfig
-              nodeWeights={this.state.nodeWeights}
-              onChange={(nw) =>
-                this.setState({nodeWeights: nw}, () => this.fire())
-              }
-            />
+            {this.pluginWeightConfigs()}
           </div>
         )}
       </React.Fragment>
     );
   }
 
+  pluginWeightConfigs() {
+    return this.props.adapters
+      .adapters()
+      .filter((x) => x.name() !== FALLBACK_NAME)
+      .map((adapter) => {
+        const onChange = (weightedTypes) => {
+          this.state.pluginNameToWeights.set(adapter.name(), weightedTypes);
+          this.fire();
+        };
+        return (
+          <PluginWeightConfig
+            key={adapter.name()}
+            adapter={adapter}
+            onChange={onChange}
+          />
+        );
+      });
+  }
+
   fire() {
-    const {edgeWeights, nodeWeights} = this.state;
+    let nodeWeights: WeightedNodeType[] = [];
+    let edgeWeights: WeightedEdgeType[] = [];
+    for (const adapter of this.props.adapters.adapters()) {
+      const weights = this.state.pluginNameToWeights.get(adapter.name());
+      const newNodeWeights =
+        weights == null
+          ? adapter.nodeTypes().map(defaultWeightedNodeType)
+          : weights.nodes;
+      const newEdgeWeights =
+        weights == null
+          ? adapter.edgeTypes().map(defaultWeightedEdgeType)
+          : weights.edges;
+      nodeWeights = nodeWeights.concat(newNodeWeights);
```

Comment:
Note that this may be quadratic in the number of plugins (or, depending
on the array implementation, may be linear). This is probably fine.